test/trento/domain/sap_system/sap_system_test.exs

```diff
@@ -26,7 +26,138 @@ defmodule Trento.SapSystemTest do
   alias Trento.Domain.SapSystem
 
   describe "SAP System registration" do
-    test "should create an incomplete SAP system aggregate and register a database instance" do
+    test "should fail when a sap system not exists and the database instance has Secondary role with system replication ACTIVE" do
+      command =
+        build(:register_database_instance_command,
+          system_replication: "Secondary",
+          system_replication_status: "ACTIVE"
+        )
+
+      assert_error(
+        command,
+        {:error, :sap_system_not_registered}
+      )
+    end
+
+    test "should fail when a sap system already exists and the database instance has Secondary role with system replication ACTIVE" do
+      sap_system_id = Faker.UUID.v4()
+      sid = Faker.StarWars.planet()
+      tenant = Faker.Beer.style()
+      instance_number = "00"
+      features = Faker.Pokemon.name()
+      host_id = Faker.UUID.v4()
+
+      initial_events = [
+        build(
+          :database_registered_event,
+          sap_system_id: sap_system_id,
+          sid: sid
+        ),
+        build(
+          :database_instance_registered_event,
+          sap_system_id: sap_system_id,
+          sid: sid,
+          tenant: tenant,
+          instance_number: "10"
+        )
+      ]
+
+      command =
+        build(
+          :register_database_instance_command,
+          sap_system_id: sap_system_id,
+          sid: sid,
+          tenant: tenant,
+          instance_number: instance_number,
+          features: features,
+          host_id: host_id,
+          system_replication: "Secondary",
+          system_replication_status: "ACTIVE"
+        )
+
+      assert_error(
+        initial_events,
+        command,
+        {:error, :sap_system_not_registered}
+      )
+    end
+
+    test "should create an incomplete SAP system aggregate and register a database instance when the system replication is disabled" do
+      sap_system_id = Faker.UUID.v4()
+      sid = Faker.StarWars.planet()
+      tenant = Faker.Beer.style()
+      instance_number = "00"
+      instance_hostname = Faker.Airports.iata()
+      features = Faker.Pokemon.name()
+      http_port = 80
+      https_port = 443
+      start_priority = "0.9"
+      host_id = Faker.UUID.v4()
+
+      assert_events_and_state(
+        [],
+        RegisterDatabaseInstance.new!(%{
+          sap_system_id: sap_system_id,
+          sid: sid,
+          tenant: tenant,
+          instance_number: instance_number,
+          instance_hostname: instance_hostname,
+          features: features,
+          http_port: http_port,
+          https_port: https_port,
+          start_priority: start_priority,
+          host_id: host_id,
+          system_replication: "Primary",
```

Comment:
In this case the system replication is enabled, as you have the `Primary` thing.
In a non SR setup, both `system_replication` and `system_replication_status` should be set to nil

Reply:
set accordingly to the test description